libshvchainpack/include/shv/chainpack/metamethod.h

```diff
@@ -92,42 +95,42 @@ class SHVCHAINPACK_DECL_EXPORT MetaMethod
 	static Signature signatureFromString(const std::string &sigstr);
 	static const char* signatureToString(Signature sig);
 	static std::string flagsToString(unsigned flags);
-	static const char* accessLevelToString(AccessLevel access_level);
+	static const char* accessLevelToAccessString(AccessLevel access_level);
+	static std::optional<AccessLevel> accessLevelFromAccessString(std::string_view s);
+	static std::optional<AccessLevel> accessLevelFromInt(int i);
+	//static AccessLevel findAccessLevel(const std::string &access);
 private:
 	std::string m_name;
 	unsigned m_flags = 0;
 	std::string m_param;
 	std::string m_result;
-	AccessLevel m_accessLevel;
+	int m_accessLevelInt;
 	RpcValue::Map m_signals;
 	std::string m_label;
 	std::string m_description;
 	RpcValue::Map m_extra;
 };
 
-struct GrantToString {
-	std::string operator()(const chainpack::MetaMethod::AccessLevel level) const {
-		return std::to_string(static_cast<int>(level));
-	}
-	std::string operator()(const std::string& role) const {
-		return role;
-	}
-};
-
-struct GrantToRpcValue {
-	chainpack::RpcValue operator()(const chainpack::MetaMethod::AccessLevel level) const {
-		return static_cast<int>(level);
-	}
-	chainpack::RpcValue operator()(const std::string& role) const {
-		return role;
-	}
-};
-
 namespace methods {
 const auto LS = MetaMethod{shv::chainpack::Rpc::METH_LS, shv::chainpack::MetaMethod::Flag::None, "LsParam", "LsResult", shv::chainpack::MetaMethod::AccessLevel::Browse};
 const auto DIR = MetaMethod{shv::chainpack::Rpc::METH_DIR, shv::chainpack::MetaMethod::Flag::None, "DirParam", "DirResult", shv::chainpack::MetaMethod::AccessLevel::Browse};
 }
 
+constexpr bool operator<(int a, const MetaMethod::AccessLevel b)
+{
+	return a < static_cast<int>(b);
+}
+
+constexpr bool operator==(int a, const MetaMethod::AccessLevel b)
+{
+	return a == static_cast<int>(b);
+}
+
+constexpr bool operator>(int a, const MetaMethod::AccessLevel b)
+{
+	return a > static_cast<int>(b);
+}
+
```

Comment:
Is this really needed? I'd rather have explicit conversions from int if possible

Reply:
AccessLevel in RpcMessage is int by definition, but it must be compared to enum on many places. Explicit static_cast<int> is possible but boring.
* brokerapp.cpp:1172
* brokerapp.cpp:1218
* brokerapp.cpp:1219
* rpcmessage.cpp:376
* rpcmessage.cpp:391

Reply:
>AccessLevel in RpcMessage is int by definition

No it's not. It's an `enum class` by definition.
>brokerapp.cpp:1172
brokerapp.cpp:1218
brokerapp.cpp:1219
rpcmessage.cpp:376
rpcmessage.cpp:391

Convert it to the enum (via the new function), and define operators for it.